### PR TITLE
[CN-702] High Availability Chaos Testing

### PR DESCRIPTION
**Pull request by semihbkgr**
## Description

Testing high-availability feature at both node and zone level. Chaos Mesh is used to simulate outage scenarios.

__Test Node Level :__

Create a GKE cluster on only one AZ and then stop the nodes in the cluster.

__Test Zone Level :__

Create a GKE cluster on multiple AZ and then stop all nodes in one specific AZ.

### CODE REVIEW

File: test/e2e/hazelcast_high_availability_test.go [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+		zone, err := zoneOfNode(ctx, droppingNodeName)
+		Expect(err).To(BeNil())
+		err = dropNodes(ctx, zone, droppingNodeName)
+		Expect(err).To(BeNil())
+
+		By("waiting the node to be dropped")
+		waitForDroppedNodes(ctx, 1)
+
+		By("checking map size after node outage")
+		nodeHzMemberPodNameMap, err := hzMemberPodsNamesInNodes(ctx, hazelcast.Name)
```
Comment (hasancelik) [inline test/e2e/hazelcast_high_availability_test.go:137]: Why are we not using exposeExternally directly?

**Review by mtyazici — COMMENTED** [review]
I think you should update Makefile to remove normal tests from running high availability tests.

**Review by hasancelik — APPROVED** [review]

File: test/e2e/client_port_forward_test.go [hunk 2]
```diff
@@ -6,14 +6,14 @@ import (
 	"strconv"
 	. "time"
 
-	hzClient "github.com/hazelcast/hazelcast-go-client"
-	hzclienttypes "github.com/hazelcast/hazelcast-go-client/types"
+	"github.com/hazelcast/hazelcast-platform-operator/internal/protocol/codec"
+	codecTypes "github.com/hazelcast/hazelcast-platform-operator/internal/protocol/types"
```
Comment (mtyazici) [inline test/e2e/client_port_forward_test.go:10]: Can you reorder the imports?

**Review by mtyazici — COMMENTED** [review]
Can we run the HA tests one time before merging?

File: Makefile [hunk 3]
```diff
@@ -205,11 +205,14 @@ test-e2e-split-kind: generate fmt vet ginkgo ## Run end-to-end tests on Kind
...
 
 test-e2e: generate fmt vet ginkgo ## Run end-to-end tests
-	USE_EXISTING_CLUSTER=true DEPLOYMENT_NAME=$(DEPLOYMENT_NAME) $(GINKGO) -r --keep-going --junit-report=test-report-${REPORT_SUFFIX}.xml --output-dir=allure-results/$(WORKFLOW_ID) --procs $(GINKGO_PARALLEL_PROCESSES) --trace --label-filter="(slow || fast) $(E2E_TEST_LABELS)" --slow-spec-threshold=100s --tags $(GO_BUILD_TAGS) --vv --progress --timeout 70m --flake-attempts 2 --output-interceptor-mode=none --coverprofile cover.out ./test/e2e -- -namespace "$(NAMESPACE)" $(GO_TEST_FLAGS)
+	USE_EXISTING_CLUSTER=true DEPLOYMENT_NAME=$(DEPLOYMENT_NAME) $(GINKGO) -r --keep-going --junit-report=test-report-$(REPORT_SUFFIX).xml --output-dir=allure-results/$(WORKFLOW_ID) --procs $(GINKGO_PARALLEL_PROCESSES) --trace --label-filter="(slow || fast) $(E2E_TEST_LABELS)" --slow-spec-threshold=100s --tags $(GO_BUILD_TAGS) --vv --progress --timeout 70m --flake-attempts 2 --output-interceptor-mode=none --coverprofile cover.out ./test/e2e -- USE_EXISTING_CLUSTER=true
```
Comment (mtyazici) [inline Makefile:208]: Why did we add USE_EXISTING_CLUSTER at the end, we have at start of the command already?

File: test/e2e/client_port_forward_test.go [hunk 4]
```diff
@@ -6,11 +6,11 @@ import (
 	"strconv"
 	. "time"
 
-	hzClient "github.com/hazelcast/hazelcast-go-client"
-	hzclienttypes "github.com/hazelcast/hazelcast-go-client/types"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
 
+	hzClient "github.com/hazelcast/hazelcast-go-client"
+	hzclienttypes "github.com/hazelcast/hazelcast-go-client/types"
```
Comment (mtyazici) [inline test/e2e/client_port_forward_test.go:13]: Can you revert the changes here?

**Review by mtyazici — APPROVED** [review]
LGTM, one thing I would suggest is to increase the number of entries for the map since we do not care about the test being fast.
